File: HRMS_Modular_Monolithic_BolierPlate Without Git/Modules/PayrollFeature/PayrollFeature.Domain/Payroll.cs
using HRMS.Core.Postgres.Common;
using HRMS.Shared.Domain.Entity;

namespace PayrollFeature.Domain
{
    public class Payroll : BaseEntity
    {
        public string? EmployeeId { get; set; }
        public string? Month { get; set; }
        public decimal? BasicSalary { get; set; }
        public decimal? Bonus { get; set; }
        public decimal? Deductions { get; set; }
        public decimal? NetSalary { get; set; }
        public string? PaymentStatus { get; set; }
        public new string DocumentType { get; set; } = nameof(Payroll);
        public UserBase? UserContext { get; set; }
    }
}

File: graphql/query/payroll.ts
import { gql } from "@apollo/client";

export const GET_ALL_PAYROLL = gql`
  query GetAllPayroll($request: GetAllPayrollRequestInput!) {
    getAllPayroll(request: $request) {
      data {
        payroll {
          payrollId
          employeeId
          month
          basicSalary
          bonus
          deductions
          netSalary
          paymentStatus
        }
      }
      message
      statusCode
      success
    }
  }
`;
